deflect/StreamSendWorker.cpp

```diff
@@ -128,6 +128,17 @@ Stream::Future StreamSendWorker::enqueueImage(const ImageWrapper& image,
         return promise.get_future();
     }
 
+    if (image.width <= 64 && image.height <= 64 &&
+        image.compressionPolicy == COMPRESSION_ON)
+    {
+        auto segment = _imageSegmenter.compressSingleSegment(image);
+        auto tasks = std::vector<Task>{
+            [this, segment] { return _sendSegment(segment); }};
+        if (finish)
+            tasks.emplace_back([this] { return _sendFinish(); });
+        return _enqueueRequest(std::move(tasks));
```

Comment:
these last 3 lines are the same as the normal case below, reuse them with an if/else for setting tasks?